### PR TITLE
Fix broken links to testgrid dashboard

### PR DESCRIPTION
**Pull request by letmerecall**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Fixes a bunch of broken link to testgrid dashboard

#### Which issue(s) this PR fixes:

This is related to an umbrella issue and fixes a task of the same:
https://github.com/kubernetes/test-infra/issues/30370

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

#### Related issues:
https://github.com/kubernetes/test-infra/issues/30370
https://github.com/kubernetes/sig-release/issues/2304

### CODE REVIEW

**Comment by letmerecall** [comment]
cc: @rjsadow

File: CHANGELOG/CHANGELOG-1.4.md
Comment (rashansmith) [inline CHANGELOG/CHANGELOG-1.4.md:1]: @letmerecall for the url: https://testgrid.k8s.io/docker the website message reads "Unknown dashboard "docker". Select a dashboard.". Is that expected?

Comment (letmerecall) [reply CHANGELOG/CHANGELOG-1.4.md:1]: I'm not super sure (I think it's expected). But this changelog is from `1.4` & there is no dashboard name 'docker' as of now, so the possibility is that it doesn't exist anymore.

Screenshot 2023-08-17 at 10 42 03 PM

Any idea who might know about this?

Comment (rjsadow) [reply CHANGELOG/CHANGELOG-1.4.md:1]: I don't think we typically update the CHANGELOGs. I would recommend reverting any updates made to them but keeping the MAINTAINERS update.

Comment (letmerecall) [reply CHANGELOG/CHANGELOG-1.4.md:1]: Got it, thanks :)

**Review by rjsadow — APPROVED** [review]

**Pull request merged**